cetlvast/suites/docs/examples/example_07_polymorphic_alloc_deleter.cpp

```diff
@@ -148,3 +231,34 @@ TEST(example_07_polymorphic_alloc_deleter, example_usage_1)
 
     //![example_usage_1]
 }
+
+TEST_F(example_07_polymorphic_alloc_deleter, example_usage_2)
+{
+    //![example_usage_2]
+
+    cetl::pmr::polymorphic_allocator<MyObject> alloc{cetl::pmr::new_delete_resource()};
+
+    auto obj0 = cetl::pmr::InterfaceFactory::make_unique<MyObject>(alloc, "obj1", 4U);
+
+    auto obj1 = cetl::pmr::InterfaceFactory::make_unique<IIdentifiable>(alloc, "obj1", 4U);
+    {
+        std::cout << "Obj1 id  : " << obj1->id() << std::endl;
+        obj1.reset();
+        std::cout << std::endl;
+    }
+    auto obj2 = cetl::pmr::InterfaceFactory::make_unique<IDescribable>(alloc, "obj2", 4U);
```

Comment:
Here I demonstrate conversion (`move` actually!) of `InterfacePtr<IDescribable>` → to compatible `InterfacePtr<INameable>`.

Reply:
No, here we have the opposite, namely upcasting (internally) from derived (aka Concrete) to a base class (aka Interface).

Reply:
Oh, I understand it now: internally, it is implemented as `static_cast<INamed>(my_object)` rather than `static_cast<INamed>(describable)`, that makes sense.